### PR TITLE
Fix Modify Headers/URL rewrite

### PR DESCRIPTION
**Pull request by LuanvP**
Fixes https://github.com/TykTechnologies/tyk-analytics/issues/663 and https://github.com/TykTechnologies/tyk/issues/1524

### CODE REVIEW

File: res_handler_header_injector_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+					"add_headers": {
+						"add-this": "header"
+					},
+					"path": "test-no-slash",
+					"method": "GET",
+					"act_on": false
+				}
+			]`), &v.ExtendedPaths.TransformResponseHeader)
+		})
+	})
```
Comment (LuanvP) [inline res_handler_header_injector_test.go]: @buger Am I setting up the middleware correctly here? It seems like it's not getting triggered for either path

json is same as for my test API

Comment (buger) [reply res_handler_header_injector_test.go]: Check if json.Unmarshal returns error, maybe it will contain some useful debugging info

**Comment by ilijabojanovic** [comment]
@buger @LuanvP 

I we tested this PR and i still see issues.

If i enable modify headers on:
headers endpoint, add/remove request/response headers working as expected
if i enable modify headers on:
/headers endpoint, add/remove request working fine, but add/remove response headers don't working at all

@buger also url-rewrite + modify headers working with when we add endpoints without backslash 
https://github.com/TykTechnologies/tyk/issues/1524

**Comment by LuanvP** [comment]
@ilijabojanovic both bugs should now be fixed

File: api_definition.go [hunk 2]
```diff
@@ -902,17 +902,31 @@ func (a *APISpec) CheckSpecMatchesStatus(r *http.Request, rxPaths []URLSpec, mod
...
+
 		// only return it it's what we are looking for
 		if !match {
 			// check for special case when using url_rewrites with transform_response
 			// and specifying the same "path" expression
-			if mode != TransformedResponse {
-				continue
-			} else if v.TransformResponseAction.Path != ctxGetUrlRewritePath(r) {
+
+			if mode == TransformedResponse {
```
Comment (buger) [inline api_definition.go]: In this case better use switch:

```
switch mode {
case TransformedResponse: 
...
case InjectHeadersResponse:
....
default:
continue 
}

Comment (LuanvP) [reply api_definition.go]: @buger done

**Comment by buger** [comment]
@ilijabojanovic pls check again

**Comment by ilijabojanovic** [comment]
@buger Verified and approved for merge

**Pull request merged**